rebar3 mechanics merged from the 'staking_contract' repo

rustaem opened the pull request.

pagdi committed:
rebar3 mechanics merged from the 'staking_contract' repo

cytadela8 commented on priv/SimpleElection.aes:
```diff
@@ -0,0 +1,455 @@
+@compiler >= 4.3
```
This contract is already present in the repository under `contracts/Election.aes`. That file should be used instead of this one

rustaem replied:
Generally, only .erl files should be present under /src
Am I to keep it under /src/contracts anyway?

cytadela8 commented on README.md:
```diff
@@ -1,6 +1,28 @@
+
+Build
+-----
+
```
Say something like:
```suggestion
Contract can be build and tested as below (see "Develop locally") or included in an rebar project. 
```

cytadela8 replied:
I mean, previously was only one way to use this repo. There was the "aeproject" javascript stuff. Now you kind of add few files to make it possible to include this as a rebar project to Erlang code. Readme should reflect that nicely. 

There are instructions to compile work with it using npm below. Change the readme so the whole text "makes sense"

pagdi committed:
Fixes

/contracts/Election.aes used instead of /priv/SimpleElection.aes;
README edited.

cytadela8 commented on LICENSE:
```diff
@@ -0,0 +1,191 @@
+                                 Apache License
```
Should use ISC Licence. You can just copy licence from core node.

cytadela8 commented on README.md:
```diff
@@ -1,6 +1,28 @@
+
+Proxy lib for the AEternity project. When used as a dependency, provides compiled and ready to use StakingContract.json
+
+The source code (in Sophia programming language) for the contract is in `/priv/SimpleElection.aes`
```
Make this a clickable link like this: `[/contarcts/Election.aes](contracts/Election.aes)`

cytadela8 reviewed (changes requested):
Change LICENCE and adjust README so it's clear that there are 2 main ways to use this repo.

pagdi committed:
LICENSE and README fixes

erlmachinedev commented:
P.S. Just keep in mind that the update of this repo will change the Core functionality and consensus dramatically. 
So I propose to keep in mind and support the right reference in the main repo https://github.com/aeternity/aeternity/blob/master/rebar.config  carefully.
